### PR TITLE
Kate/handling multiple attributes and speed up detection split

### PR DESCRIPTION
**Pull request by jihyeonyi**
### Summary
While handling multi-attributes, we've found that task-specific splitters are mishandling them.
This fix includes,
1. better handling of the multi-attributes case 
2. speed up & OOM issue is fixed for the detection_split

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by jihyeonyi** [comment]
I've added the code to remove the subset name restriction for classification and detection.

File: datumaro/plugins/splitter.py [hunk 1]
```diff
@@ -70,21 +74,29 @@ def _get_uniq_annotations(dataset):
...
-        if valid is None:
-            valid = ["train", "val", "test"]
+        subsets = set()
+        valid = ["train", "val", "test"]
+        # remove subset name restriction
+        # regarding https://github.com/openvinotoolkit/datumaro/issues/194
         for subset, ratio in splits:
-            assert subset in valid, \
-                "Subset name must be one of %s, but got %s" % (valid, subset)
+            if restrict:
```
Comment (zhiltsov-max) [inline datumaro/plugins/splitter.py:85]: What is the point of this restriction? Is it possible to require more than 3 subsets? Do subsets have special processing depending on the name?

File: datumaro/plugins/splitter.py [hunk 2]
```diff
@@ -129,37 +152,74 @@ def _group_by_attr(items):
...
+
+        # float--> numerical, others(int, string, bool) --> categorical
+        def _is_number(value):
+            if isinstance(value, str):
+                try:
+                    float(value)
+                    return True
+                except ValueError:
+                    return False
+            elif isinstance(value, float):
```
Comment (zhiltsov-max) [inline datumaro/plugins/splitter.py:164]:
```suggestion
            elif isinstance(value, (int, float)):
```

Comment (zhiltsov-max) [reply datumaro/plugins/splitter.py:164]: Check if you can get any use for `datumaro.util.cast`.

File: datumaro/plugins/splitter.py (hunk 1 above)
Comment (jihyeonyi) [reply datumaro/plugins/splitter.py:85]: if restrict is Ture, the subset names are restricted into ["train", "val", "test"]. This is for re-identification at the moment.
For detection and classification, this restriction isn't needed so I changed this code. This is regarding issue #194

File: datumaro/plugins/splitter.py (hunk 2 above)
Comment (jihyeonyi) [reply datumaro/plugins/splitter.py:164]: 'int' is regarded as categorical at the moment. 
That's why I checked 'float' only. 
For the `datumaro.util.cast`, I'll check it.

**Review by zhiltsov-max — DISMISSED** [review]

**Pull request merged**